lib/config.js

```diff
@@ -0,0 +1,105 @@
+"use strict";
+
+/**
+ * Support for configuration file.
+ */
+
+// requires
+const Log = require("log");
+const {
+	Confinode,
+	Level,
+	anyItem,
+	booleanItem,
+	choiceItem,
+	defaultValue,
+	dictionary,
+	literal,
+	numberItem,
+	optional,
+	stringItem
+} = require("confinode");
+
+// globals
+const log = new Log("info");
+
+/**
+ * Load configuration from file.
+ */
+exports.loadConfig = function(options) {
+	if (options.debug) {
+		log.level = Log.DEBUG;
+	}
+	if (options.quiet) {
+		log.level = Log.NOTICE;
+	}
+	const description = literal({
+		server: literal({
+			delay: numberItem(0),
+			error: numberItem(0),
+			percent: stringItem("")
+		}),
+		limits: literal({
+			maxRequests: numberItem(0),
+			concurrency: numberItem(1),
+			maxSeconds: numberItem(0),
+			timeout: optional(stringItem())
+		}),
+		request: literal({
+			method: choiceItem(["GET", "POST", "PUT", "DELETE", "PATCH", "get", "post", "put", "delete", "patch"], "GET"),
+			contentType: stringItem(""),
+			body: defaultValue(anyItem(), ""),
+			file: stringItem(""),
+			cookies: defaultValue(new CookieDescription(), []),
+			headers: defaultValue(dictionary(stringItem()), {})
+		}),
+		ssl: literal({
+			secureProtocol: stringItem(""),
+			insecure: booleanItem(),
+			cert: stringItem(""),
+			key: stringItem("")
+		}),
+		requestGenerator: stringItem(""),
+		recover: booleanItem(true),
+		agentKeepAlive: booleanItem(),
+		proxy: stringItem(""),
+		requestsPerSecond: numberItem(0),
+		indexParam: stringItem("")
+	});
+	const confinode = new Confinode("loadtest", description, { logger, mode: "sync" });
+	const result = confinode.search();
+	return result ? result.configuration : {};
+};
+
+/**
+ * Logging function of confinode.
+ */
+function logger(msg) {
+	const fn =
```

Comment:
This structure is a bit odd.

Reply:
I'll make it clearer using a switch…case